### PR TITLE
chore: helm dep update for lint + lint manifest_staging

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Update deps for helm lint
- Enable helm lint for `manifest_staging`

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Please answer the following questions with yes/no**:

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

**Special Notes for Reviewers**:

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -336,3 +336,7 @@ ASALocalRun/
 /_output*/
 /_output
 .DS_Store
+
+# helm chart dependencies
+charts/csi-secrets-store-provider-azure/charts/
+manifest_staging/charts/csi-secrets-store-provider-azure/charts/
```
Comment (ritazh) [inline .gitignore:342]: Wouldn't this impact us from making updates to the yamls in manifest_staging in the future?

Comment (aramase) [reply .gitignore:342]: The charts directory nested inside the parent charts dir is for dependencies. We never store the driver package in the repo.

Comment (ritazh) [reply .gitignore:342]: Got it!

**Review by ritazh — APPROVED** [review]
lgtm

**Pull request merged**